logic/src/androidMain/kotlin/com/wire/kalium/logic/data/message/MessageRepositoryExtensions.kt

```diff
@@ -0,0 +1,43 @@
+package com.wire.kalium.logic.data.message
+
+import androidx.paging.PagingConfig
+import androidx.paging.PagingData
+import androidx.paging.map
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.id.IdMapper
+import com.wire.kalium.persistence.dao.message.KaliumPager
+import com.wire.kalium.persistence.dao.message.MessageDAO
+import com.wire.kalium.persistence.dao.message.MessageEntity
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+actual interface MessageRepositoryExtensions {
+    suspend fun getPaginatedMessagesByConversationIdAndVisibility(
+        conversationId: ConversationId,
+        visibility: List<Message.Visibility>,
+        pagingConfig: PagingConfig,
+    ): Flow<PagingData<Message>>
+}
+
+actual class MessageRepositoryExtensionsImpl actual constructor(
+    private val messageDAO: MessageDAO,
+    private val idMapper: IdMapper,
+    private val messageMapper: MessageMapper,
+) : MessageRepositoryExtensions {
+
+    override suspend fun getPaginatedMessagesByConversationIdAndVisibility(
+        conversationId: ConversationId,
+        visibility: List<Message.Visibility>,
+        pagingConfig: PagingConfig,
+    ): Flow<PagingData<Message>> {
+        val pager: KaliumPager<com.wire.kalium.persistence.Message, MessageEntity> = messageDAO.platformExtensions.getPagerForConversation(
```

Comment:
Could this be imported at the top?

Reply:
Only with an `alias`, because `com.wire.kalium.data.message.Message` is on the same package as this file and is also used, see line 32: `): Flow<PagingData<Message>> {`

As it's a single usage on the whole file, introducing an `alias` wouldn't solve much anyway 🤷🏼‍♂️